src/ubx.cpp

```diff
@@ -1282,13 +1282,75 @@ GPSDriverUBX::payloadRxAddNavSat(const uint8_t b)
 				// Part 2 complete: decode Part 2 buffer
 				unsigned sat_index = (_rx_payload_index - sizeof(ubx_payload_rx_nav_sat_part1_t)) /
 						     sizeof(ubx_payload_rx_nav_sat_part2_t);
-				_satellite_info->svid[sat_index]	  = static_cast<uint8_t>(_buf.payload_rx_nav_sat_part2.svId);
+
+				// convert gnssId:svId to a 8 bit number (UBX svId from NAV-SVINFO, NMEA would need at last 9 bytes)
```

Comment:
I guess you meant:
```suggestion
				// convert gnssId:svId to a 8 bit number (UBX svId from NAV-SVINFO, NMEA would need at least 9 bits)
```
And can you add where the numbering is coming from?

Reply:
Yes, it's a typo. I can also remove that bit of information.
I had the issue, that the NAV-SAT uses two 8 bit fields to define the satellite number, the GNSS ID (as "namespace") and the satellite's number (inside the namespace).
But satellite_info only has one 8 bit field. So there are 2 possible solutions: using the numbering scheme from the old NAV-SVINFO message or the NMEA numbering. But the NMEA numbering has numbers greater than 255, so it wouldn't fit. Therefore I used the "old" numbering.

Reply:
Yes I understood that, and is fine for now, just wanted to make sure the comment makes that clear as well.